pkg/pathutils/pathutils.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+Not licensed for re-use.
+*/
+
+package pathutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func IsValidFile(path string) bool {
```

Comment:
`IsValidFile` does not represent well what this function does. `IsFile` is probably sufficient. The `Valid` word implies it does some sort of validation, which it does not. Alternatively, you can probe if the file is readable/writable:

```
func IsFileReadable(path string) bool {
  const R_OK uint32 = 4
  return syscall.Access(path, R_OK) != syscall.EACCES
}
```

For reference, the second argument to `syscall.Access` can be copied from the C code `unistd.h`:

```
#define	R_OK	4		/* Test for read permission.  */
#define	W_OK	2		/* Test for write permission.  */
#define	X_OK	1		/* Test for execute permission.  */
#define	F_OK	0		/* Test for existence.  */
```

You see, `F_OK` can be used for verifying if it exists or not too.

Reply:
Nice suggestion. I don't think probing whether we have r/w access to the file is necessary, as we are using this helper to perform a simple check if a file exists.

Reply:
Alright. Doing the `syscall.Access` may make this whole package easier to test.

Reply:
I just tried out your suggestion, but doesn't seem to build even after importing `syscall`. I think this is an os-dependent API, and according to https://pkg.go.dev/syscall, it seems like `syscall` is deprecated. Their suggestion is to use the packages here https://pkg.go.dev/golang.org/x/sys according to the platform that we're targetting.

Since we want our CLI to be cross-compatible, we probably shouldn't overcomplicate this with platform-specific syscall branches.